### PR TITLE
sms branching

### PR DESCRIPTION
**Pull request by pascalpp**
- refactor of SMS dispatch logic to allow for commands and flows
- also adds Issues to web form and SMS flow

### CODE REVIEW

File: src/controller/sms/signup/signup_steps.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+const containsYes = /(yes|yeah|ya)/i
+
+function isAffirmative(response) {
+  if (startsWithY.test(response)) return true
+  if (containsYes.test(response)) return true
+  return false
+}
+
+function validateZip(zip) {
+  return validZip.test(zip) && zip
```
Comment (mattstauffer) [inline src/controller/sms/signup/signup_steps.js:32]: what's the `&& zip`?

File: src/controller/sms/signup/signup_steps.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
+/* eslint-disable no-param-reassign */
+const message = require('../message')
+const saveUser = require('../../../user/save_user')
+const issues = require('../../../public/issues').issues
+const events = require('../schedule')
+const debug = require('debug')('sms')
+
+function notStarted(req, step) {
```
Comment (mattstauffer) [inline src/controller/sms/signup/signup_steps.js:8]: I assume these methods will be extracted later, once we have multiple flows (not mad it's all here for now, just double checking)

**Review by mattstauffer — APPROVED** [review]

File: src/controller/sms/signup/signup_steps.js (hunk 2 above)
Comment (pascalpp) [reply src/controller/sms/signup/signup_steps.js:8]: yeah if other flows use a similar control mechanism. a flow with more branching (choose your own adventure) might need different helpers.

File: src/controller/sms/signup/signup_steps.js (hunk 1 above)
Comment (pascalpp) [reply src/controller/sms/signup/signup_steps.js:32]: this function returns the zip if it's valid, or false if it's not. so the `&& zip` just changes a `true` return value to the zip itself.

**Pull request merged**